server/AuthController.js

```diff
@@ -12,6 +13,42 @@ module.exports = {
     res.set(JWT.renewTokenHeader, token);
     res.status(200).send({ token })
   },
+  impersonate: (_grabUser, id='id') => async (req, res) => {
+    try {
+      let token = JWT.issue({
+        // The new user will be whatever the target id is
+        user: req.param(id),
+        // Add the previous user to the back of the stack of users ids
+        // Makes it easy to use with lodash :D
+        impersonateStack: _.concat(req.user.id)(req.tokenPayload.impersonateStack || [])
+      })
+      res.set(JWT.renewTokenHeader, token)
+      res.status(200).send({token})
+    } catch (e) {
+      // Return 500 because there was a problem with the server
+      // Either grabbing permissions failed (developer's fault)
+      // Or generating a token failed (server's fault)
+      res.status(500).send({message: e.message || e})
+    }
+  },
+  impersonatePop: async (req, res) => {
+    try {
+      let tokenRaw = {
+        // Grab last user of the stack and make the new user
+        user: _.last(req.tokenPayload.impersonateStack),
+        // New stack should be the all but the last user
+        // since they're the new user
+        impersonateStack: _.dropRight(1)(req.tokenPayload.impersonateStack)
+      }
+      // _.dropRight will still return an empty array if it dropped all users
+      if (_.isEmpty(tokenRaw.impersonateStack)) delete tokenRaw.impersonateStack
+      let token = JWT.issue(tokenRaw)
+      res.set(JWT.renewTokenHeader, token)
```

Comment:
This isn't actionable right now, but this seems like a recurring thing in this library that we may want to factor out